### PR TITLE
Add ModalHeader

### PR DESCRIPTION
**Pull request by dherault**
Before: 
```jsx
<Modal>content</Modal>
```

After:
```jsx
<Modal><ModalHeader>Title</ModalHeader>content</Modal>
```

So I've added the new ModalHeader component.

### CODE REVIEW

File: src/stories/Modal.stories.tsx [hunk 1]
```diff
@@ -29,47 +19,27 @@ function Template2(args: any) {
       <Modal
         open={open}
         onClose={() => setOpen(false)}
-        {...args}
-      />
+      >
+        <ModalHeader onClose={() => setOpen(false)}>
```
Comment (michaeljguarino) [inline src/stories/Modal.stories.tsx:23]: what value does splitting out this component provide?

Comment (dherault) [reply src/stories/Modal.stories.tsx:23]: More control over the customization of the modal

Comment (dherault) [reply src/stories/Modal.stories.tsx:23]: Also, we already have a separate `ModalActions`, so it's in the continuation.

Comment (michaeljguarino) [reply src/stories/Modal.stories.tsx:23]: Will it break all existing code?  Or is the title prop and onClose still supported on modal itself

File: src/components/ModalHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      >
+        {children}
+      </P>
+      {typeof onClose === 'function' && (
+        <Flex
+          align="center"
+          justify="center"
+          padding="small"
+          margin={-12}
+          borderRadius="50%"
```
Comment (michaeljguarino) [inline src/components/ModalHeader.tsx:38]: I'm pretty sure we shouldn't round this border actually, probably use something like a 2-3px radius.  We can confirm with nick but it looks clunky

**Pull request merged**